### PR TITLE
PUBDEV-5637 Add Standard Error of Mean (SEM) to PDP

### PR DESCRIPTION
**Pull request by navdeep-G**
This PR adds standard error of the mean to PDP

Standard error of mean versus standard deviation:

In scientific and technical literature, experimental data are often summarized either using the mean and standard deviation of the sample data or the mean with the standard error. This often leads to confusion about their interchangeability. However, the mean and standard deviation are descriptive statistics, whereas the standard error of the mean is descriptive of the random sampling process. The standard deviation of the sample data is a description of the variation in measurements, while the standard error of the mean is a probabilistic statement about how the sample size will provide a better bound on estimates of the population mean, in light of the central limit theorem.

Put simply, the standard error of the sample mean is an estimate of how far the sample mean is likely to be from the population mean, whereas the standard deviation of the sample is the degree to which individuals within the sample differ from the sample mean. If the population standard deviation is finite, the standard error of the mean of the sample will tend to zero with increasing sample size, because the estimate of the population mean will improve, while the standard deviation of the sample will tend to approximate the population standard deviation as the sample size increases.

Another reference: http://www.mas.ncl.ac.uk/~njnsm/medfac/docs/se&ci.pdf

### CODE REVIEW

File: h2o-core/src/main/java/hex/PartialDependence.java [hunk 1]
```diff
@@ -128,9 +129,11 @@ public void compute2() {
                 if (_model_id.get()._output.nclasses() == 2) {
                   meanResponse[which] = preds.vec(2).mean();
                   stddevResponse[which] = preds.vec(2).sigma();
+                  stdErrorofTheMeanResponse[which] = preds.vec(2).sigma()/ preds.numRows();
```
Comment (mstensmo) [inline h2o-core/src/main/java/hex/PartialDependence.java]: Shouldn't it be sqrt? SEM = SD / sqrt(n)

File: h2o-core/src/main/java/hex/PartialDependence.java [hunk 2]
```diff
@@ -128,9 +129,11 @@ public void compute2() {
...
                 } else if (_model_id.get()._output.nclasses() == 1) {
                   meanResponse[which] = preds.vec(0).mean();
                   stddevResponse[which] = preds.vec(0).sigma();
+                  stdErrorofTheMeanResponse[which] = preds.vec(0).sigma()/ preds.numRows();
```
Comment (mstensmo) [inline h2o-core/src/main/java/hex/PartialDependence.java]: same here

File: h2o-core/src/main/java/hex/PartialDependence.java [hunk 3]
```diff
@@ -106,6 +106,7 @@ public void compute2() {
         Futures fs = new Futures();
         final double meanResponse[] = new double[colVals.length];
         final double stddevResponse[] = new double[colVals.length];
+        final double stdErrorofTheMeanResponse[] = new double[colVals.length];
```
Comment (mstensmo) [inline h2o-core/src/main/java/hex/PartialDependence.java]: Maybe  stdErrorOfTheMeanResponse if it's camel case

File: h2o-core/src/main/java/hex/PartialDependence.java (hunk 1 above)
Comment (navdeep-G) [reply h2o-core/src/main/java/hex/PartialDependence.java]: whoops. i thought i changed that.

File: h2o-core/src/main/java/hex/PartialDependence.java (hunk 3 above)
Comment (mstensmo) [reply h2o-core/src/main/java/hex/PartialDependence.java]: or maybe just SEMResponse

Comment (navdeep-G) [reply h2o-core/src/main/java/hex/PartialDependence.java]: yep, can make that change
